device_doctor/lib/src/list_processes.dart

```diff
@@ -0,0 +1,84 @@
+// Copyright 2020 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:io';
+
+import 'package:meta/meta.dart';
+import 'package:path/path.dart' as path;
+
+import 'process_helper.dart';
+import 'utils.dart';
+
+/// Returns a list of Process IDs (PIDs).
+Future<List<int>> listFlutterProcessIds(Directory flutterDirectory) {
+  if (Platform.isMacOS || Platform.isLinux) {
+    return _listFlutterProcessesPosix(flutterDirectory);
+  }
+  if (Platform.isWindows) {
+    return _listFlutterProcessesWindows(flutterDirectory);
+  }
+  throw 'Unsupported platform: ${Platform.operatingSystem}';
+}
+
+// POSIX implementation
```

Comment:
Not sure where we will be using this functionality from the context that the tool will be running separately from a task.